brainiak/utils/fmrisim.py

```diff
@@ -1595,6 +1603,66 @@ def _generate_noise_temporal_drift(trs,
         phase = (timepoints / (trs - 1) * cycles * 2 * np.pi) + phaseshift
         noise_drift = np.sin(phase)
 
+    elif basis == 'cos_power_drop':
+
+        # Make a vector counting each TR
+        timepoints = np.linspace(0, trs - 1, trs) * tr_duration
+
+        # Specify the other timing information
+        duration = trs * tr_duration
+
+        # How bases do you have?
+        basis_funcs = int(np.floor(2 * duration))
+
+        noise_drift = np.zeros((timepoints.shape[0], basis_funcs))
+        for basis_counter in list(range(1, basis_funcs + 1)):
+            # What steps do you want to take for this basis function
+            random_phase = np.random.rand() * np.pi * 2
+
+            timepoint_phase = (timepoints / duration * np.pi * basis_counter)
+
+            # In radians, what is the value for each time point
+            timepoints_basis = timepoint_phase + random_phase
+
+            # Store the drift from this basis func
+            noise_drift[:, basis_counter - 1] = np.cos(timepoints_basis)
+
+        def power_drop(r, L, F):
+            # Function to return the drop rate for the power of basis functions
+            # In other words, how much should the weight of each basis function
+            # reduce in order to make the power you retain be above a 0.99
+            # r is the power reduction rate which should be between 0 and 1
+            # L is the duration of the run in seconds
+            # F is period of the cycle in seconds
+
+            percent_retained = 0.99  # What is the percentage of drift retained
+            numerator = 1 - r ** (4 * L / F)  # Power of this period
```

Comment:
In this line, at least `L / F` is unitless, corresponding to the number of periods within the duration. So that makes sense to serve as the power that `r` is raised to (see my comment below).
This to me looks like something as the amount of reduction of power from the lowest frequency to the highest frequency. If `demoninator` is something like the amount of reduction in the power of starting frequency (from some baseline), then you are trying to say that from this starting frequency to the highest frequency, we have only dropped to 0.99. Is this correct? This does not seem to resemble what I would imagine as "power you retain" although I am also not so clear of the meaning of "retain" here. But I think maybe I misunderstood some part. Please feel free to correct me.

Reply:
After having a second thought, I think now I understand what you mean by retain. It may be fine, but some more explanation may help anyone else reading the code. Maybe you can complete your original sentence as "how much should the weight of each basis function reduce in order to make the power you retain up to the frequency with its period as F be above 99% of the total power up to the highest frequency" (although I think we have different ideas of what should be set as the highest frequency here, see my other comments)

Reply:
Good suggestion, added it with some light rewording